service/indexer/indexer_test.go

```diff
@@ -810,34 +901,35 @@ var _ = Describe("Indexer Test", func() {
 			mockStoreManager.On("UpdateBlocks", mock.Anything, newBlocks[15:19], [][]*types.Receipt{{receipt}, {receipt}, {receipt}, {receipt}}, [][]*types.TransferLog{nilTransferLogs, nilTransferLogs, nilTransferLogs, nilTransferLogs}, &model.Reorg{
 				From:     15,
 				FromHash: blocks[15].Hash().Bytes(),
-				To:       17,
-				ToHash:   blocks[17].Hash().Bytes(),
+				To:       15,
+				ToHash:   blocks[15].Hash().Bytes(),
 			}).Return(nil).Once()
 
-			var recvCh chan<- *types.Header
-			recvCh = ch
-			mockEthClient.On("SubscribeNewHead", mock.Anything, recvCh).Return(mockSub, nil).Once()
+			for i, c := range mockEthClients {
+				c.On("SubscribeNewHead", mock.Anything, mock.Anything).Return(subFunc(i), nil).Once()
+			}
 
 			go func() {
-				ch <- newBlocks[18].Header()
+				time.Sleep(time.Second)
+				chs[0] <- newBlocks[18].Header()
 				time.Sleep(time.Second)
 				cancel()
 			}()
 
-			err := idx.Listen(ctx, ch, 0)
+			err := idx.Listen(ctx, 0)
 			Expect(err).Should(Equal(context.Canceled))
 		})
 	})
 
-	Context("Listen() old block with Reorg", func() {
+	Context("Listen() reorg with older header", func() {
```

Comment:
add a test that have headers from different eth client?

Reply:
OK